tests/integration/helpers/cluster.py

```diff
@@ -106,10 +106,17 @@ def retry_exception(num, delay, func, exception=Exception, *args, **kwargs):
         return
     raise StopIteration('Function did not finished successfully')
 
-def subprocess_check_call(args, detach=False, nothrow=False):
+def subprocess_check_call(args, detach=False, nothrow=False, num_tries=1):
     # Uncomment for debugging
     #logging.info('run:' + ' '.join(args))
-    return run_and_check(args, detach=detach, nothrow=nothrow)
+    for i in range(num_tries):
+        try:
+            return run_and_check(args, detach=detach, nothrow=nothrow)
+        except Exception as ex:
+            if i == num_tries - 1:
+                raise ex
+            else:
+                logging.info("Got execption while executing command: %s", ex)
```

Comment:
There is already `retry_exception()`, and I think adding delay between retries may help a little.

Reply:
Though it does not show proper exception:
https://github.com/ClickHouse/ClickHouse/blob/e5c538f1575242d2ed958e80192fe13704413b91/tests/integration/helpers/cluster.py#L107